File: stack_using_linke_list.c
#include<stdio.h>
#include<stdlib.h>
struct node
{
    int data;
    struct node *next;
};

struct node *head = NULL;

void push()
{
	struct node *newNode = malloc(sizeof(struct node));
	int val;
	printf("enter the value to be push : ");
	scanf("%d",&val);
    	newNode->data = val;
	newNode->next = head;
	head = newNode;
	printf("value after inserted\n");
	show();
}

void pop()
{
	struct node *temp;
	if(head == NULL)
        	printf("Stack is Empty\n");
    	else
    	{
        	printf("Poped element = %d\n", head->data);
                temp = head;
	     	head = head->next;
	        free(temp);
		printf("value after pop:\n");
		show();
    	}
}
void isempty()
{
	if(head == NULL)
		printf("true\n");
	else
		printf("false\n");
}

void show()
{
	struct node *temp = head;
	while(temp != NULL)
    	{
        	printf("%d->", temp->data);
        	temp = temp->next;
    	}
    	printf("NULL\n");
}
int choice=0;
int main()
{
	while(choice != 5)  
    	{  
        	printf("Chose one from the below options...\n");  
        	printf("\n1.Push\n2.Pop\n3.Show\n4.isempty\n5.Exit");  
        	printf("\n Enter your choice \n");        
        	scanf("%d",&choice);  
        	switch(choice)  
        	{
			case 1:
			{   
                		push();  
                		continue;
			}  
			case 2:  
            		{  
                		pop();  
                		continue; 
            		}  
            		case 3:  
            		{  
                		show();  
                		continue; 
            		}  
                	case 4:  
            		{  
                		isempty();  
                		continue; 
			}               		
            		case 5:   
            		{  
                		printf("Exiting....");  
                		break;   
            		}  
            		default:  
            		{  
                		printf("Please Enter valid choice ");  
            		}   
        	}
	}
   	return 0;
}

File: stack_using_linke_list.java
import static java.lang.System.exit;

class stack_using_linke_list
{
	public static void main(String[] args)
	{
		
		StackUsingLinkedlist obj= new StackUsingLinkedlist();
		int choice=0;  
        	Scanner sc = new Scanner(System.in);   
        	System.out.println("*********Stack operations using array*********\n");  
        	System.out.println("\n------------------------------------------------\n");  
        	while(choice != 4)  
        	{  
            		System.out.println("\nChose one from the below options...\n");  
            		System.out.println("\n1.Push\n2.Pop\n3.Show\n4.Exit");  
            		System.out.println("\n Enter your choice \n");        
            		choice = sc.nextInt();  
            		switch(choice)  
            		{  
                		case 1:  
                		{   
                    			s.push(sc);  
                    			continue; 
                		}  
                		case 2:  
                		{  
                    			s.pop();  
                    			continue; 
                		}  
                		case 3:  
                		{  
                    			s.display();  
                    			continue; 
                		}  
                		case 4:   
                		{  
                    			System.out.println("Exiting....");  
                    			break;   
                		}  
                		default:  
                		{  
                    			System.out.println("Please Enter valid choice ");  
                		}   
			}
	}

}
class StackUsingLinkedlist 
{
	private class Node 
	{
		int data;
		Node link; 
	}
	Node top;

	// Constructor

	StackUsingLinkedlist() 
	{ 
		this.top = null; 
	}

	
	public void push(int x) 
	{
		Node temp = new Node();
		if (temp == null) 
		{
			System.out.print("\nHeap Overflow");
			return;
		}

		temp.data = x;

		temp.link = top;

		top = temp;
	}

	public boolean isEmpty() 
	{ 
		return top == null; 
	}

	public int topvalue()
	{
		
		if (!isEmpty()) 
		{
			return top.data;
		}
		else 
		{
			System.out.println("Stack is empty");
			return -1;
		}
	}

	public void pop() 
	{
		if (top == null) 
		{
			System.out.print("\nStack Underflow");
			return;
		}
		top = (top).link;
	}

	public void display()
	{
		if (top == null) 
		{
			System.out.printf("\nStack Underflow");
		}
		else 
		{
			Node temp = top;
			while (temp != null) 
			{
				System.out.print(temp.data);
				temp = temp.link;
				if(temp != null)
					System.out.print(" -> ");
			}
		}
	}
}
